include/bluetooth/mesh/mgmt.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2019 Tobias Svehagen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_BLUETOOTH_MESH_MGMT_H_
+#define ZEPHYR_INCLUDE_BLUETOOTH_MESH_MGMT_H_
+
+#if defined(CONFIG_BT_MESH_NETWORK_MGMT)
+#define NODE_COUNT    CONFIG_BT_MESH_MGMT_NODE_COUNT
+#define SUBNET_COUNT  CONFIG_BT_MESH_MGMT_SUBNET_COUNT
+#define APP_KEY_COUNT CONFIG_BT_MESH_MGMT_APP_KEY_COUNT
+#else
+#define NODE_COUNT    0
+#define SUBNET_COUNT  0
+#define APP_KEY_COUNT 0
+#endif
+
+struct bt_mesh_mgmt_node {
+	u16_t addr;
+	u16_t net_idx;
+	u8_t  dev_key[16];
+	u8_t  num_elem;
+};
+
+struct bt_mesh_mgmt_subnet {
+	u16_t net_idx;
+
+	bool kr_flag;
+	u8_t kr_phase;
+
+	struct {
+		u8_t net_key[16];
+	} keys[2];
+};
+
+struct bt_mesh_mgmt_app_key {
+	u16_t net_idx;
+	u16_t app_idx;
+	struct {
+		u8_t id;
+		u8_t key[16];
+	} keys[2];
+};
+
+enum {
+	BT_MESH_DB_VALID,
+	BT_MESH_DB_NET_PENDING,
+	BT_MESH_DB_KEYS_PENDING,
+	BT_MESH_DB_NODES_PENDING,
+
+
+	/* Don't touch - intentionally last */
+	BT_MESH_DB_FLAG_COUNT,
+};
+
+struct bt_mesh_mgmt_net {
+	u32_t iv_index;
+
+	ATOMIC_DEFINE(flags, BT_MESH_DB_FLAG_COUNT);
+
+	struct bt_mesh_mgmt_node nodes[NODE_COUNT];
+	struct bt_mesh_mgmt_subnet subnets[SUBNET_COUNT];
+	struct bt_mesh_mgmt_app_key app_keys[APP_KEY_COUNT];
+};
+
+extern struct bt_mesh_mgmt_net bt_mesh_network;
+
+struct bt_mesh_mgmt_subnet *bt_mesh_mgmt_subnet_alloc(u16_t net_idx);
+void bt_mesh_mgmt_subnet_del(struct bt_mesh_mgmt_subnet *sub, bool store);
+struct bt_mesh_mgmt_subnet *bt_mesh_mgmt_subnet_get(u16_t idx);
+
+struct bt_mesh_mgmt_node *bt_mesh_mgmt_node_alloc(u16_t addr, u8_t num_elem,
+						  u16_t net_idx);
+void bt_mesh_mgmt_node_del(struct bt_mesh_mgmt_node *node, bool store);
+struct bt_mesh_mgmt_node *bt_mesh_mgmt_node_find(u16_t addr);
```

Comment:
Should probably be `bt_mesh_mgmt_node_get` to align with subnet. Both will do a search.